dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1115,6 +1116,27 @@ void TeachingTip::OnActionButtonClicked(const winrt::IInspectable&, const winrt:
 
 void TeachingTip::OnPopupOpened(const winrt::IInspectable&, const winrt::IInspectable&)
 {
+    if (winrt::IUIElement10 uiElement10 = *this)
+    {
+        if (auto xamlRoot = uiElement10.XamlRoot())
+        {
+            m_currentXamlRootSize = xamlRoot.Size();
+            if (m_xamlRootChangedToken)
+            {
+                xamlRoot.Changed(m_xamlRootChangedToken);
+            }
+            m_xamlRootChangedToken = xamlRoot.Changed({ this, &TeachingTip::XamlRootChanged });
+        }
+    }
+    else
+    {
+        if (auto coreWindow = winrt::CoreWindow::GetForCurrentThread())
+        {
+            m_windowSizeChangedRevoker.revoke();
+            m_windowSizeChangedRevoker = coreWindow.SizeChanged(winrt::auto_revoke, { this, &TeachingTip::WindowSizeChanged });
```

Comment:
assigning this below will revoke before assignment, this explicit .revoke() shouldn't be necessary.